packages/dbml-core/src/parse/dbml/src/compiler.ts

```diff
@@ -1,52 +1,535 @@
+    }),
+    tokens: this.createQuery(
+      Query.Parse_Tokens,
+      (): Readonly<SyntaxToken>[] => this.parse._().getValue().tokens,
+    ),
+    rawDb: this.createQuery(
+      Query.Parse_RawDb,
+      (): Readonly<Database> | undefined => this.parse._().getValue().rawDb,
+    ),
+    publicSymbolTable: this.createQuery(
+      Query.Parse_PublicSymbolTable,
+      (): Readonly<SymbolTable> => this.parse._().getValue().ast.symbol!.symbolTable!,
+    ),
+  };
+
+  // Find the stack of nodes/tokens, with the latter being nested inside the former
+  // that contains `offset`
+  containers = this.createQuery(
+    Query.Containers,
+    (offset: number): readonly Readonly<SyntaxNode>[] => {
+      let curNode: Readonly<SyntaxNode> = this.parse.ast();
+      const res: SyntaxNode[] = [curNode];
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
+        const memberChain = getMemberChain(curNode);
+        const foundMem = memberChain.find((mem) => isOffsetWithinSpan(offset, mem));
+        if (foundMem === undefined || foundMem instanceof SyntaxToken) {
+          break;
+        }
+        res.push(foundMem);
+        curNode = foundMem;
+      }
+
+      return res;
+    },
+  );
+
+  // A namespace for symbol-related queries
+  readonly symbol = {
+    ofName: this.createQuery(
+      Query.Symbol_OfName,
+      (
+        nameStack: string[],
+      ): readonly Readonly<{ symbol: NodeSymbol; kind: SymbolKind; name: string }>[] => {
+        const { symbolTable } = this.parse.ast().symbol!;
+        let currentPossibleSymbolTables: SymbolTable[] = [symbolTable!];
+        let currentPossibleSymbols: { symbol: NodeSymbol; kind: SymbolKind; name: string }[] = [];
+        // eslint-disable-next-line no-restricted-syntax
+        for (const name of nameStack) {
+          currentPossibleSymbols = currentPossibleSymbolTables.flatMap((st) =>
+            generatePossibleIndexes(name).flatMap((index) => {
+              const symbol = st.get(index);
+              const res = destructureIndex(index).unwrap_or(undefined);
+
+              return !symbol || !res ? [] : { ...res, symbol };
+            }));
+          currentPossibleSymbolTables = currentPossibleSymbols.flatMap((e) =>
+            (e.symbol.symbolTable ? e.symbol.symbolTable : []));
+        }
+
+        return currentPossibleSymbols;
+      },
+    ),
+    members: this.createQuery(
+      Query.Symbol_Members,
+      (
+        ownerSymbol: NodeSymbol,
+      ): readonly Readonly<{ symbol: NodeSymbol; kind: SymbolKind; name: string }>[] =>
+        (ownerSymbol.symbolTable ?
+          [...ownerSymbol.symbolTable.entries()].map(([index, symbol]) => ({
+              ...destructureIndex(index).unwrap(),
+              symbol,
+            })) :
+          []),
+    ),
+  };
+
+  // Return all possible symbols corresponding to a stack of name
+  membersOfName = this.createQuery(
+    Query.MembersOfName,
+    (
+      nameStack: string[],
```

Comment:
This implies the name stack should always originate from the outer ast scope. What if we want to find members for name declared in some local scope? 

```
Table users {
  Enum status {
    ok,
    failed
  }

  status.[suggestions]
}
```

Reply:
Yeah, `membersOfName` should be called something like `publicMembersOfName`. To handle this case, we have the `symbol.members` query though, which is passed the symbol of `status` and use the symbol table in that symbol to do the lookup.